ros2topic/ros2topic/verb/pub.py

```diff
@@ -117,6 +152,66 @@ def main(args):
             args.keep_alive)
 
 
+def choose_qos(node, args):
+
+    if (args.qos_profile is not None or
+        args.qos_reliability is not None or
+        args.qos_durability is not None or
+        args.qos_depth is not None or
+        args.qos_history is not None):
+
+        if args.qos_profile is None:
+            args.qos_profile = default_profile_str
+        return qos_profile_from_short_keys(args.qos_profile,
+                                           reliability=args.qos_reliability,
+                                           durability=args.qos_durability,
+                                           depth=args.qos_depth,
+                                           history=args.qos_history)
+
+    qos_profile = QoSPresetProfiles.get_from_short_key(default_profile_str)
+    reliability_reliable_endpoints_count = 0
+    durability_transient_local_endpoints_count = 0
+
+    subs_info = node.get_subscriptions_info_by_topic(args.topic_name)
+    subscribers_count = len(pubs_info)
+    if subscribers_count == 0:
+        return qos_profile
+
+    for info in subs_info:
+        if (info.qos_profile.reliability == QoSReliabilityPolicy.RELIABLE):
+            reliability_reliable_endpoints_count += 1
+        if (info.qos_profile.durability == QoSDurabilityPolicy.TRANSIENT_LOCAL):
+            durability_transient_local_endpoints_count += 1
```

Comment:
Same here, there isn't any benefit of doing this.
Using TRANSIENT_LOCAL as the default will match both VOLATILE and TRANSIENT_LOCAL subscriptions.

Reply:
In the case of transient local, the publisher actually might need to keep messages for a longer time than when using volatile.
Using it as the default here for purposes of endpoint compatibility doesn't hurt, particularly considering that we're using a KEEP_LAST history by default.